src/ml/surrogate.hpp

```diff
@@ -248,7 +291,7 @@ class SurrogateModel
     DBG(Surrogate, "Generating new model under (%s)", model_path);
     std::shared_ptr<SurrogateModel<TypeInValue>> torch_model =
         std::shared_ptr<SurrogateModel<TypeInValue>>(
-            new SurrogateModel<TypeInValue>(model_path, is_cpu));
+            new SurrogateModel<TypeInValue>(model_path, is_cpu, is_DeltaUQ));
```

Comment:
Can you check and raise an error if there is an existing instance of the model with a different `is_DeltaUQ` flag value?

Reply:
Done